src/melee/ft/ftcoll.c

```diff
@@ -179,3 +192,3028 @@ f32 func_800765F0(Fighter* fp, HSD_GObj* victim,
+    this_fp->unk_grab_val = F32_MAX;
+    victim_gobj = lbl_804D782C->x20_fighters;
+
+    while (victim_gobj != NULL) {
+        if (func_80086FD4(this_gobj, victim_gobj) == false) {
+            victim_fp = victim_gobj->user_data;
+            if (((!(victim_fp->x2219_flag.bits.b1)) &&
+                 ((!victim_fp->x222A_flag.bits.b0) &&
+                  ((func_8016B168() == false) || (func_8016B0D4() != false) ||
+                   ((u8) victim_fp->x61B_team != (u8) this_fp->x61B_team)) &&
+                  ((s32) victim_fp->x1988 == 0) &&
+                  ((s32) victim_fp->x198C == 0) &&
+                  (!(victim_fp->x221D_flag.bits.b6) &&
+                   !(victim_fp->x1A6A & this_fp->x1A68) &&
+                   !((victim_fp->x2224_flag.bits.b2))))))
+            {
+                for (i = 0; i < 4; i++) {
+                    this_hit = ftHitGetPtr(this_fp, i);
+                    if ((this_hit->tangiblity != Vulnerable) &&
+                        (this_hit->element == (u32) HitElement_Catch) &&
+                        ((((this_hit->x40_b2) &&
+                           (victim_fp->xE0_ground_or_air == GA_Air)) ||
+                          ((((u8) this_hit->x40_b3) &&
+                            ((s32) victim_fp->xE0_ground_or_air ==
+                             GA_Ground))))))
+                    {
+                        if (func_8000ACFC(victim_fp, this_hit) == false) {
+                            for (j = 0; j < victim_fp->x119E_hurtboxNum; j++) {
+                                if ((s32) victim_fp->x11A0_fighterHurtbox[j]
+                                        .x48_is_grabbable != false)
+                                {
+                                    if (func_80007ECC(
+                                            this_hit,
+                                            &victim_fp->x11A0_fighterHurtbox[j],
+                                            func_8007F804(victim_fp),
+                                            this_fp->x34_scale.y,
+                                            victim_fp->x34_scale.y,
+                                            victim_fp->xB0_pos.z) != 0)
+                                    {
+                                        if (func_80084CE4(this_fp, victim_fp) ==
+                                            false)
+                                        {
+                                            func_80076808(this_fp, this_hit, 0,
+                                                          victim_fp, 0);
+
+                                            ftGrabDist(this_fp, victim_fp);
+                                        }
+                                        goto next_gobj;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    next_gobj:
+        victim_gobj = victim_gobj->next;
+    }
+}
+
+#endif
```

Comment:
Function is here. Needs a lot of cleanup but so does a lot of code in the codebase. Matches except that it adds two entries to `sdata2`.